### ISSUE TITLE
How to define panic_fmt is not documented well

### ISSUE THREAD

**Issue by habnabit** [rust-lang/rust#33677]
This includes on <https://doc.rust-lang.org/book/no-stdlib.html> and <https://doc.rust-lang.org/core/>. The former has an incorrect signature on `panic_fmt` and the latter doesn't mention that you need a `#[lang]` annotation. The relationship between `panic_fmt` and `rust_begin_unwind` (the former is a lang item, and the latter is a symbol that the former produces) has been confusing people on `#rust` recently, and should probably be documented as well.

For reference, the correct way to define it is:

```rust
#[lang="panic_fmt"]
#[unwind]
extern fn panic_fmt(args: ::core::fmt::Arguments, file: &'static str, line: u32) -> ! {
    // ...
}
```

Every part of that signature (except the function name) is necessary; any omission can cause unclear or misleading errors, or lead to the function being called with the wrong ABI.

**Comment by oli-obk** [comment]
related: #9307

**Comment by steveklabnik** [comment]
Sigh, I just ran into this today. Will send a PR tomorrow.

**Comment by steveklabnik** [comment]
apparently tomorrow never came. I am willing to mentor anyone who wants to add this

**Comment by pepyakin** [comment]
It seems https://doc.rust-lang.org/book/no-stdlib.html is no longer exists
So what is preferred way of declaring `panic_fmt` lang item?

**Comment by steveklabnik** [comment]
https://doc.rust-lang.org/unstable-book/language-features/lang-items.html#writing-an-executable-without-stdlib

**Comment by pepyakin** [comment]
Thanks!
